ingestion/src/metadata/ingestion/source/database/redshift/queries.py

```diff
@@ -31,12 +31,16 @@
           AND starttime < '{end_time}'
           LIMIT {result_limit}
   ),
+  deduped_querytext AS ( -- In some instances, rows are duplicated, causing LISTAGG to fail due to exceeding the maximum str length of 65535.
+    SELECT DISTINCT *
+    FROM pg_catalog.stl_querytext
+  ),
   full_queries AS (
     SELECT
           query,
           LISTAGG(CASE WHEN LEN(RTRIM(text)) = 0 THEN text ELSE RTRIM(text) END, '')
             WITHIN GROUP (ORDER BY sequence) AS query_text
-      FROM pg_catalog.stl_querytext
+      FROM deduped_querytext
```

Comment:
@pmbrull need your opinion on this, I was thinking that instead of doing a `SELECT DISTINCT *
    FROM pg_catalog.stl_querytext` here IMO we should directly use `queries` here which filtered result-set  
```suggestion
      FROM queries
```

Reply:
The `queries` CTE uses a different table to the `full_queries` CTE. I did a join to reduce the size of the deduplicated CTE so that the filters get applied to both queries. Let me know what you think! 🙌

Reply:
you're right, my bad.

LGTM then, thanks for taking the time for this